driver/src/contracts/stablex_contract.rs

```diff
@@ -0,0 +1,138 @@
+#[cfg(test)]
+extern crate mock_it;
+
+use web3::types::{H256, U128, U256};
+
+use crate::error::DriverError;
+
+use super::base_contract::BaseContract;
+
+type Result<T> = std::result::Result<T, DriverError>;
+
+#[allow(dead_code)] // event_loop needs to be retained to keep web3 connection open
+struct StableXContractImpl {
+    contract: BaseContract
+}
+
+#[allow(dead_code)] // event_loop needs to be retained to keep web3 connection open
+impl StableXContractImpl {
+    pub fn new(contract: BaseContract) -> Self {
+        // Should we assert that the contract is indeed a StableX contract?
+        StableXContractImpl {
+            contract
+        }
+    }
+}
+
+pub trait StableXContract {
+    fn get_current_auction_index(&self) -> Result<U256>;
+    // TODO - auction_data should parse and return relevant orders and account balances.
+    fn get_auction_data(&self, _index: u32) -> Result<H256>;
+
+    fn submit_solution(
+        &self,
+        _batch_index: u32,
```

Comment:
I think we should either have both batch_index types as u32 or U256 (`get_current_auction_index ()`). Not sure which one makes more sense. @josojo what was your thought around this? I can see it being mixed in the contract as well (e.g. [here as U256](https://github.com/gnosis/dex-contracts/blob/master/contracts/StablecoinConverter.sol#L148) and [here as u32](https://github.com/gnosis/dex-contracts/blob/master/contracts/StablecoinConverter.sol#L162)).

Reply:
We could easily avoid the following error by using either of `U128` or `U256`

```
self.base.contract
            .query("getCurrentStateIndex", (), None, Options::default(), None)
            .wait()
             ^^^^ the trait `web3::contract::tokens::Tokenizable` is not implemented for `u32`
```

I have also noticed a place in the contract where `uint` is used instead of `uint32`... surprised this does cause any solidity errors.

Reply:
I'm fine using U256 everywhere. Can you adjust the smart contracts as well to be consistent?